spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -42,6 +44,41 @@
 MODULES_FAQ_URL = (
     "https://docs.spyder-ide.org/5/faq.html#using-packages-installer")
 
+# Required version of Spyder-kernels
+SPYDER_KERNELS_MIN_VERSION = '2.3.0'
+SPYDER_KERNELS_MAX_VERSION = '2.4.0'
+SPYDER_KERNELS_VERSION = (
+    f'>={SPYDER_KERNELS_MIN_VERSION};<{SPYDER_KERNELS_MAX_VERSION}')
+SPYDER_KERNELS_VERSION_MSG = _(
+    '>= {0} and < {1}').format(
+        SPYDER_KERNELS_MIN_VERSION, SPYDER_KERNELS_MAX_VERSION)
+SPYDER_KERNELS_CONDA = (
+    f'conda install spyder&#45;kernels={SPYDER_KERNELS_MIN_VERSION[:-2]}')
+SPYDER_KERNELS_PIP = (
+    f'pip install spyder&#45;kernels=={SPYDER_KERNELS_MIN_VERSION[:-1]}*')
+
+ERROR_SPYDER_KERNEL_VERSION = _(
+    "The Python environment or installation whose "
+    "interpreter is located at"
+    "<pre>"
+    "    <tt>{0}</tt>"
+    "</pre>"
+    "doesn't have the right version of <tt>spyder-kernels</tt> "
+    " installed ({1} instead of {2}). "
+    "Without this module is not possible for Spyder to "
+    "create a console for you.<br><br>"
+    "You can install it by activating your environment (if "
+    "necessary) and then running in a system terminal:"
+    "<pre>"
+    "    <tt>{3}</tt>"
+    "</pre>"
+    "or"
+    "<pre>"
+    "    <tt>{4}</tt>"
+    "</pre>"
```

Comment:
```suggestion
    "The Python environment or installation whose interpreter is located at"
    "<pre>"
    "    <tt>{0}</tt>"
    "</pre>"
    "doesn't have the right version of <tt>spyder-kernels</tt> installed ({1} "
    "instead of {2}). Without this module is not possible for Spyder to "
    "create a console for you.<br><br>"
    "You can install it by activating your environment (if necessary) and "
    "then running in a system terminal:"
    "<pre>"
    "    <tt>{3}</tt>"
    "</pre>"
    "or"
    "<pre>"
    "    <tt>{4}</tt>"
    "</pre>"
```

Improve formatting.